src/main/java/core/basesyntax/service/date/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.date;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private Map<String, Integer> storage;
+
+    public ReportGeneratorImpl(Map<String, Integer> storage) {
+        this.storage = storage;
+    }
+
+    public ReportGeneratorImpl() {
+    }
```

Comment:
Avoid creating empty constructors if they are not required. In this case, the empty constructor allows creating an instance of `ReportGeneratorImpl` without providing a `storage` which can lead to a NullPointerException when calling the `getReport` method. Therefore, the empty constructor should be removed.